clang/lib/DPCT/PatternRewriter.cpp

```diff
@@ -481,106 +473,53 @@ static std::optional<MatchResult> findFullMatch(const MatchPattern &Pattern,
         return {};
       }
 
-      if (PatternIndex == 0 && Index - 1 >= 0 &&
-          isIdentifiedChar(Input[Index - 1]) &&
-          isIdentifiedChar(Input[Index])) {
-        return {};
-      }
-
-      // If input value has been matched to the end but match pattern template
-      // still has value, it is considered not matched case.
-      if (Index == Size - 1 && PatternIndex < PatternSize - 1) {
-        return {};
-      }
-
-      // If match pattern template has been matched to the end but input value
-      // still not the end, it is considered not matched case.
-      if (PatternIndex == PatternSize - 1 &&
-          isIdentifiedChar(Input[Index + 1])) {
-        return {};
-      }
-
-      if (PatternIndex == PatternSize - 1 &&
-          !isIdentifiedChar(Input[Index + 1])) {
-        Result.FullMatchFound = true;
-      }
-
-      Index++;
-      PatternIndex++;
-      continue;
-    }
-
-    if (std::holds_alternative<CodeElement>(Element)) {
-      const auto &Code = std::get<CodeElement>(Element);
-      MatchPattern Suffix(Pattern.begin() + PatternIndex + 1,
-                          Pattern.begin() + PatternIndex + 1 +
-                              Code.SuffixLength);
-
-      int Next = parseCodeElement(Suffix, Input, Index, false);
-      if (Next == -1) {
-        return {};
-      }
-
-      std::string ElementContents = Input.substr(Index, Next - Index);
-      if (SrcFileType == SourceFileType::SFT_CMakeScript) {
-        if (Code.Name == "empty" && !ElementContents.empty() &&
-            ElementContents.find_first_not_of(' ') != std::string::npos) {
-          // For reversed variable ${empty}, it should be empty string or string
-          // only including spaces.
+      if (Mode == RuleMatchMode::Full) {
+        if (PatternIndex == 0 && Index - 1 >= 0 &&
```

Comment:
the code is similar to full mode can we extract it to a function?

Reply:
Okay, will update it in the next update.